pkg/chunk/cached_store.go

```diff
@@ -651,14 +676,24 @@ func (store *cachedStore) load(key string, page *Page, cache bool, forceCache bo
 	if used > SlowRequest {
 		logger.Infof("slow request: GET %s (%v, %.3fs)", key, err, used.Seconds())
 	}
-	if store.downLimit != nil && compressed {
-		store.downLimit.Wait(int64(n))
-	}
 	store.objectDataBytes.WithLabelValues("GET").Add(float64(n))
 	store.objectReqsHistogram.WithLabelValues("GET").Observe(used.Seconds())
 	if err != nil {
 		store.objectReqErrors.Add(1)
+		logger.Error("get %s: %s", key, err)
+	}
+	var err1 error
+	if store.downLimit != nil && compressed {
```

Comment:
there is a race between UpdateCachedStoreRateLimit and here.

Reply:
1. fixbug rate
2. golang.org/x/time/rate allows update limit or burst, can ensure concurrency.
3. It returns an error if n exceeds the Limiter's burst size, the Context is canceled, or the expected wait time exceeds the Context's Deadline.
cancel and timeout are not set here, so only n(token count) is larger than burst size, can cause error. To ensure the effect of rate.limiter, this case should not be executed normally.
4. about metrics, make tokens at a constant speed(rate.NewLimiter), we may prefer to know whether there are config exception. if it is normal, block until get the needed tokens.